src/annotator/anchoring/pdf.js

```diff
@@ -295,21 +305,24 @@ async function anchorByPosition(pageIndex, start, end) {
     page.textLayer.renderingDone
   ) {
     // The page has been rendered. Locate the position in the text layer.
-    const root = page.textLayer.textLayerDiv;
-
-    // Do a sanity check to verify that the page text extracted by `getPageTextContent`
-    // matches the transparent text layer.
     //
-    // See https://github.com/hypothesis/client/issues/3674.
-    if (pageText !== root.textContent) {
-      /* istanbul ignore next */
-      warnOnce(
-        'PDF text layer content does not match page text. This will cause anchoring misalignment.'
-      );
-    }
+    // We allow for differences in whitespace between the text returned by
+    // `getPageTextContent` and the text layer content. Any other differences
+    // will cause mis-anchoring.
 
-    const startPos = new TextPosition(root, start);
-    const endPos = new TextPosition(root, end);
+    const root = page.textLayer.textLayerDiv;
+    const textLayerStr = /** @type {string} */ (root.textContent);
+
+    const [textLayerStart, textLayerEnd] = translateOffsets(
+      pageText,
+      textLayerStr,
```

Comment:
> I'm a little less clear on the provenance of the textContent of the text-layer root div. Presumably this is pulled from the PDF document's own text layer and chunked into positioned span elements. Are we to go on the assumption that the textContent of this div is reliably the textual content of the PDF? Does PDF.js do any manipulation or transformation of this textual content)? Put another way: is the textual content of this text-layer div subject to non-negligible change across PDF.js versions or do we see it as generally constant?

The `textContent` of the text layer root is ultimately constructed from the output of the same text extraction API that we use here, and the mapping of that text into spans is pretty direct. However what happened here, and could happen in future, is that PDF.js calls the text extraction API with different arguments than we do and that altered the whitespace in the output. Previously we tried to adapt our calls to the API so that we called the text extraction API with the same arguments and filtered the results in the same way that PDF.js does, but this complicates maintenance over the long term. 

To put it another way, I think of the contents of the text layer div as being a version of what `getPageTextContent` returns that has been corrupted by adding or removing whitespace in arbitrary locations.

> There are some places in the code that make unqualified references to "the page's text" (e.g. the docstring for this method, the start and end args).

References to "the page's text" generally refer to what `getPageTextContent` returns.

Reply:
> However what happened here, and could happen in future, is that PDF.js calls the text extraction API with different arguments than we do and that altered the whitespace in the output. Previously we tried to adapt our calls to the API so that we called the text extraction API with the same arguments and filtered the results in the same way that PDF.js does, but this complicates maintenance over the long term.

OK, this was really key. Is it possible to somehow get this context explained in comments or  commit metadata?